nutils/evaluable.py

```diff
@@ -217,6 +222,24 @@ def _compile_expression(self, py_self: _pyast.Variable, *args: _pyast.Expression
 
         return py_self.get_attr('evalf').call(*args)
 
+    def argument_degree(self, argument):
+        '''return the heighest power of argument of self is polynomial,
+        or raise NotPolynomal otherwise.'''
+        # IMPORTANT: since we are tracking only the highest power, we cannot
+        # ever lower a power, e.g. via division. To see this, consider the sum
+        # of a 0th and 1st power, registered as degree 1. This would be lowered
+        # via division to 0, thereby suggesting that the evaluable is constant
+        # while in reality it is not polynomial.
+        if argument not in self.arguments:
```

Comment:
I don't think this is an actual problem, but the arguments of the shape of `ArrayFromTuple` are not included in the `ArrayFromTuple.arguments`, which renders this test invalid if `argument` is part of the shape, but not part of the tuple. The proper fix is to include the shape args in `ArrayFromTuple.arguments`.

Reply:
The ArrayFromTuple shape can only depend on an argument if the evaluable itself also depends on that argument, otherwise it could not evaluate to an array of the correct shape, so the default implementation should suffice.